src/System.Private.CoreLib/shared/System/Text/Rune.cs

```diff
@@ -662,15 +662,42 @@ public static OperationStatus DecodeUtf8FromEnd(ReadOnlySpan<byte> utf8Source, o
             }
         }
 
-        // returns the number of chars written
-        private int EncodeToUtf16(Span<char> destination)
+        /// <summary>
+        /// Encodes this <see cref="Rune"/> to a UTF-16 destination buffer.
+        /// </summary>
+        /// <param name="destination">The buffer to which to write this value as UTF-16.</param>
+        /// <returns>The number of <see cref="char"/>s written to <paramref name="destination"/>.</returns>
+        /// <exception cref="ArgumentException">
+        /// If <paramref name="destination"/> is not large enough to hold the output.
+        /// </exception>
+        public int Encode(Span<char> destination)
         {
-            Debug.Assert(destination.Length >= Utf16SequenceLength, "Caller should've provided a large enough buffer.");
-            bool success = TryEncode(destination, out int charsWritten);
-            Debug.Assert(success, "TryEncode should never fail given a large enough buffer.");
+            if (!TryEncode(destination, out int charsWritten))
+            {
+                ThrowHelper.ThrowArgumentException_DestinationTooShort();
```

Comment:
It looks like ThrowArgumentException_DestinationTooShort doesn't include "destination" as a parameter name in the exception that gets thrown.  Can we fix that, too?

Reply:
Yes. It looks like every place where it's called the parameter name is indeed _destination_, so we may as well make the change.